server/players.py

```diff
@@ -66,21 +66,19 @@ def __init__(
 
         self.state = PlayerState.IDLE
 
-        self.faction = 1
-
-        self._lobby_connection = lambda: None
         if lobby_connection is not None:
             self.lobby_connection = lobby_connection
 
-        self._game = lambda: None
-        self._game_connection = lambda: None
+    lobby_connection = WeakAttribute["LobbyConnection"]()
+    game = WeakAttribute["Game"]()
+    game_connection = WeakAttribute["GameConnection"]()
```

Comment:
I would have put these before the `__init__`?

Reply:
Yea maybe that's better, I wasn't sure where to put them since they're replacing the property definitions and those were after init.